android/Omnivore/app/src/main/java/app/omnivore/omnivore/feature/save/SaveSheetActivity.kt

```diff
@@ -133,28 +135,25 @@ class SaveSheetActivity : AppCompatActivity() {
     }
 
     private fun WorkManager.enqueueSaveWorker(context: Context, url: String) {
-        val constraints = Constraints.Builder()
-            .setRequiredNetworkType(NetworkType.CONNECTED)
-            .build()
-
-        val syncWorkerRequest = OneTimeWorkRequest.Builder(LibrarySyncWorker::class.java)
-            .setConstraints(constraints)
-            .addTag(url)
-            .build()
+        val saveData = workDataOf("url" to url)
 
-        val inputData = Data.Builder()
-            .putString("url", url)
+        val saveWork = OneTimeWorkRequestBuilder<SaveURLWorker>()
+            .setInputData(saveData)
+            .setConstraints(Constraints.Builder()
+                .setRequiredNetworkType(NetworkType.CONNECTED)
+                .build())
             .build()
 
-        val saveURLWorkRequest = OneTimeWorkRequest.Builder(SaveURLWorker::class.java)
-            .setConstraints(constraints)
-            .setInputData(inputData)
-            .setExpedited(OutOfQuotaPolicy.RUN_AS_NON_EXPEDITED_WORK_REQUEST)
-            .addTag(url)
+        val syncWork = OneTimeWorkRequestBuilder<LibrarySyncWorker>()
+            .setConstraints(Constraints.Builder()
+                .setRequiredNetworkType(NetworkType.CONNECTED)
+                .build())
+            .setInitialDelay(5, TimeUnit.SECONDS)
             .build()
 
-        beginWith(saveURLWorkRequest)
-            .then(syncWorkerRequest)
+        WorkManager.getInstance(context)
```

Comment:
This is not needed btw, this is an Extension funtion that uses the WorkManager instance already created in onCreate.
So it can be like this without it:
```
beginUniqueWork("saveAndSync", ExistingWorkPolicy.REPLACE, saveWork)
            .then(syncWork)
            .enqueue()
```
also the context parameter can be removed